core/modules/wmgr/python/dbMgr.py

```diff
@@ -604,8 +639,91 @@ def locadData(dbName, tblName, chunkId=None):
+    if chunkId is not None:
+        if request.path.endswith('/overlap'):
+            tblName = tblName + 'FullOverlap_' + str(chunkId)
+        else:
+            tblName = tblName + '_' + str(chunkId)
+        if not dbConn.tableExists(tblName, dbName):
+            raise ExceptionResponse(404, "ChunkTableMissing",
+                                    "Chunk table %s.%s does not exist" % (dbName, tblName))
+
+    # optional load-options
+    options = dict(delimiter='\t', enclose='"', escape='\\', terminate='\n',
+                   compressed=None, local="0")
+    formOptions = request.form.get('load-options')
+    _log.debug('formOptions: %s', formOptions)
+    if formOptions:
+        formOptions = url_decode(formOptions)
+
+        # check that there are no non-recognized options
+        for key, value in formOptions.items():
+            if key not in options:
+                raise ExceptionResponse(400, "IllegalOption",
+                                        "unrecognized option %s in load-options" % key)
+            options[key] = value
+        _log.debug('options: %s', options)
+
+    # get table data
+    data = request.files.get('table-data')
+    if data is None:
+        raise ExceptionResponse(400, "MissingFileData", "table-data part is missing from request")
+    _log.debug('data: name=%s filename=%s', data.name, data.filename)
+
+    # named pipe to send data to mysql, make it in a temporary dir
+    tmpDir = tempfile.mkdtemp()
+    fifoName = os.path.join(tmpDir, 'tabledata.dat')
+    os.mkfifo(fifoName, 0600)
+
+    # do we need to uncompress?
+    compressed = _getArgFlag(options, 'compressed', None)
+    if compressed is None:
+        compressed = data.filename.endswith('.gz')
+
+    _log.debug('starting data copy thread')
+    copyThread = Thread(target=_copy, args=(data, fifoName, compressed))
+    copyThread.start()
+
+    try:
+
+        # Build the query, we use LOCAL data loading to avoid messing with grants and
+        # file protection.
+        sql = "LOAD DATA LOCAL INFILE %(file)s INTO TABLE {0}.{1}".format(dbName, tblName)
+        sql += " FIELDS TERMINATED BY %(delimiter)s ENCLOSED BY %(enclose)s \
+                 ESCAPED BY %(escape)s LINES TERMINATED BY %(terminate)s"
+
+        del options['compressed']
+        options['file'] = fifoName
+
+        # execute query
+        _log.debug("query: %s, data: %s", sql, options)
+        cursor = dbConn._conn.cursor()
+        cursor.execute(sql, options)
+        count = cursor.rowcount
+
+    finally:
+
+        # cleanup for thread
+        _log.debug('joining data copy thread')
+        copyThread.join()
+
+        # do not need file any more
+        data.close()
+
+        # do not forget to delete temporary stuff
+        try:
+            _log.info('deleting temporary directory %s', tmpDir)
+            shutil.rmtree(tmpDir)
+        except Exception as exc:
+            _log.warning('failed to delete temporary directory %s: %s', tmpDir, exc)
```

Comment:
I think this is reasonable, but it may be worth considering the use of context managers for cleanup (so that as much as possible gets cleaned up if you do run into odd exceptions, e.g. inside this finally clause).